### PR TITLE
Update to Grape

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
This updates Scrypto to `grape-d605b3a`.
(it includes [and thus supersedes] https://github.com/radixdlt/babylon-node/pull/611, which we didn't manage to merge in time :D)

### CODE REVIEW

File: core-rust/state-manager/src/jni/test_state_reader.rs [hunk 1]
```diff
@@ -112,7 +112,7 @@ pub enum TransactionOutcomeJava {
 #[derive(Debug, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
 pub struct JavaValidatorInfo {
     pub stake_unit_resource: ResourceAddress,
-    pub unstake_receipt_resource: ResourceAddress,
+    pub claim_receipt_resource: ResourceAddress,
```
Comment (LukasGasior1) [inline core-rust/state-manager/src/jni/test_state_reader.rs:115]: not sure why we have `receipt` in the name here?

File: core/src/test/java/com/radixdlt/api/core/NetworkSubmitTransactionTest.java [hunk 2]
```diff
@@ -239,7 +239,8 @@ public void test_valid_but_rejected_transaction_should_be_rejected() throws Exce
       assertThat(rejectedDetails.getIsFresh()).isTrue();
       assertThat(rejectedDetails.getErrorMessage())
           .isEqualTo(
-              "ErrorBeforeFeeLoanRepaid(SystemModuleError(CostingError(FeeReserveError(LoanRepaymentFailed))))");
+              "ErrorBeforeFeeLoanRepaid(SystemModuleError(CostingError(FeeReserveError(LoanRepaymentFailed"
+                  + " { xrd_owed: 0.00847164 }))))");
```
Comment (LukasGasior1) [inline core/src/test/java/com/radixdlt/api/core/NetworkSubmitTransactionTest.java:243]: this test couldn't be any more specific :smile:

File: core-rust/core-api-server/core-api-schema.yaml [hunk 3]
```diff
@@ -1676,18 +1718,47 @@ components:
...
+        to_validator_set:
+          type: string
+          description: |
+            The string-encoded decimal representing the amount of fee in XRD paid to the validator set.
+            A decimal is formed of some signed integer `m` of attos (`10^(-18)`) units, where `-2^(256 - 1) <= m < 2^(256 - 1)`.
+        to_burn:
+          type: string
+          description: |
+            The string-encoded decimal representing the amount of fee burnt, in XRD.
+            A decimal is formed of some signed integer `m` of attos (`10^(-18)`) units, where `-2^(256 - 1) <= m < 2^(256 - 1)`.
```
Comment (LukasGasior1) [inline core-rust/core-api-server/core-api-schema.yaml:1743]: we'll need to update all those comments :)

Comment (LukasGasior1) [reply core-rust/core-api-server/core-api-schema.yaml:1743]: _(note to self mostly)_

File: core-rust/state-manager/src/jni/test_state_reader.rs (hunk 1 above)
Comment (jakrawcz-rdx) [reply core-rust/state-manager/src/jni/test_state_reader.rs:115]: actually needs to through a PR https://github.com/radixdlt/babylon-node/pull/616

File: common/src/main/java/com/radixdlt/utils/UInt192.java [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+    for (int i = 1; i < numbers.length; i++) {
+      r = lcm(r, numbers[i]);
+      if (r == null || r.compareTo(cap) > 0) {
+        return null;
+      }
+    }
+    return r;
+  }
+
+  private static UInt192 lcm(UInt192 x, UInt192 y) {
```
Comment (LukasGasior1) [inline common/src/main/java/com/radixdlt/utils/UInt192.java:374]: and these methods come from `UInt256s`

File: common/src/main/java/com/radixdlt/utils/UInt64.java [hunk 5]
```diff
@@ -118,10 +128,55 @@ private UInt64(Long underlyingValue) {
     this.underlyingValue = Objects.requireNonNull(underlyingValue);
   }
 
+  public UInt64 add(UInt64 other) {
+    return new UInt64(underlyingValue + other.underlyingValue);
```
Comment (LukasGasior1) [inline common/src/main/java/com/radixdlt/utils/UInt64.java:132]: note from author: regular arithmetic for add/sub "just works" with unsigned values

**Review by malexandru-rdx — APPROVED** [review]
LGTM!

File: common/src/main/java/com/radixdlt/utils/UInt64.java [hunk 6]
```diff
@@ -118,10 +128,55 @@ private UInt64(Long underlyingValue) {
...
+  public boolean isOdd() {
+    return underlyingValue % 2 == 1;
+  }
+
+  public UInt64 shiftLeft() {
+    return new UInt64(underlyingValue << 1);
+  }
+
+  public UInt64 shiftRight() {
+    return new UInt64(underlyingValue >> 1);
```
Comment (jakrawcz-rdx) [inline common/src/main/java/com/radixdlt/utils/UInt64.java:152]: The `>>` copies the "sign"; you want `>>>`.

**Review by jakrawcz-rdx — COMMENTED** [review]
My comments are fixed by https://github.com/radixdlt/babylon-node/pull/621 (which I will immediately merge here, using my new special red button).

**Comment by jakrawcz-rdx** [comment]
As agreed with Łukasz on Slack, I reviewed the latest `Decimal` changes he contributed here, and I am merging this.

**Pull request merged**